### PR TITLE
Get a script's redeem script. Implements #7

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by JBaczuk** [comment]
Should probably rebase with master

**Review by JBaczuk — CHANGES_REQUESTED** [review]
Sorry, I'm trying to follow, can you explain what this does and why we need it?

**Comment by wardlem** [comment]
> Sorry, I'm trying to follow, can you explain what this does and why we need it?

This is to get the redeemScript (as a parsed script object) from a p2sh script sig.

**Comment by wardlem** [comment]
@JBaczuk To clarify the motivation for this PR, let us consider the case where this module is being used in a block explorer to display the ASM of a script.  If a P2SH scriptSig is detected, the UI could choose to also display the redeem script.  For example:

```
Script Pub Key: [0014983c113b3b34d109e72aa6d4246e2430b3240161]
Redeem Script: OP_0 [14983c113b3b34d109e72aa6d4246e2430b3240161]
```

**Comment by JBaczuk** [comment]
Looks like this needs to be rebased/conflict resolved

**Comment by wardlem** [comment]
@JBaczuk This has been rebased.

File: src/bscript.js [hunk 1]
```diff
@@ -84,6 +86,21 @@ class BScript {
     }
   }
 
+  get redeemScript () {
+    if (this.tokens.length === 1 && this.tokens[0].type === Token.LITERAL) {
```
Comment (JBaczuk) [inline src/bscript.js]: Why does this.tokens.length have to be 1, and in what cases would it not be?

**Review by JBaczuk — APPROVED** [review]
I didn't notice any problems, but have a question

File: src/bscript.js (hunk 1 above)
Comment (wardlem) [reply src/bscript.js]: The script must be a p2sh scriptSig, which is always a single PUSH_DATA op.  That is why the length must be 1 and have type `Token.LITERAL`.

**Pull request merged**